<a href="https://colab.research.google.com/github/Celes-ovo/Python/blob/main/3%EC%9E%A5_%EC%A0%95%EB%A6%AC.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#<파이썬 3장 내용 정리>

#1. abs(absolute value : 절댓값)

In [ ]:
x = abs(-7) + abs(3.3)
x

10.3

In [ ]:
a = abs(-3.3)
a

3.3

#2. pow
###pow(power)


power에 수가 2개 있을 때와 3개 있을 때의 차이를 보자

* 먼저 수가 2개일 때.

In [ ]:
pow(2, 4)

16

2 ** 4와 같음을 볼 수 있다.

* 다음으로 수가 3개일 때.

In [ ]:
pow(2, 4, 5)

1

In [ ]:
help(pow)

Help on built-in function pow in module builtins:

pow(x, y, z=None, /)
    Equivalent to x**y (with two arguments) or x**y % z (with three arguments)
    
    Some types, such as ints, are able to use a more efficient algorithm when
    invoked using the three argument form.



> 위 설명에 따르면, pow는 세 개의 수일 때에는 x**y % z의 형태로 동작한다.
>>따라서 (2**4)%3 이기 때문에 1이 나옴.

#3. round
###round(반올림)

In [ ]:
round(1.45)

1

In [ ]:
round(11.23345678, 4)

11.2335

숫자 하나만 있을 때에는 그 수의 정수 반올림 값, 숫자가 두개일 때에는 앞의 수를 소숫점을 반올림해 뒤의 수만큼의 자리로 표기.

> 예를 들어, 위의 코드의 경우 뒤의 수가 4이므로 4자리까지 반올림해 표기하였다.

# + abs, round, pow가 같이 있는 경우.

>abs들이 각각 계산되고, +가 그 이후 처리됨.

In [ ]:
y = pow(abs(-2), round(4.3))
y

16

> pow(abs(-2), round(4.3)) 수식에서
  abs(-2), round(4.3)가 각각 먼저 계산됨.

     이때 abs, round 순으로 계산됨!!   

  abs(-2) = 2, round(4.3) = 4

  이후 pow(2, 4) = 16

#4. help

아주아주 소중한 기능.    함수가 무엇인지 알고 싶을 때 설명을 읽을 수 있는 기능.

In [ ]:
help(abs)

Help on built-in function abs in module builtins:

abs(x, /)
    Return the absolute value of the argument.



In [ ]:
help(round)

Help on built-in function round in module builtins:

round(...)
    round(number[, ndigits]) -> number
    
    Round a number to a given precision in decimal digits (default 0 digits).
    This returns an int when called with one argument, otherwise the
    same type as the number. ndigits may be negative.



이 함수가 무엇인지, 함수의 설명이 나옴을 볼 수 있다!

# + int나 float 등은 함수로도 사용될 수 있다.

In [ ]:
int(34.6)

34

In [ ]:
int(-4.3)

-4

In [ ]:
float(21)


21.0

> int와 float가 각각의 의미를 유지하면서 함수의 역할도 같이 해 줌을 볼 수 있다.

#5. id

어떤 object의 메모리 주소를 볼 수 있는 함수

In [ ]:
help(id)

Help on built-in function id in module builtins:

id(obj, /)
    Return the identity of an object.
    
    This is guaranteed to be unique among simultaneously existing objects.
    (CPython uses the object's memory address.)



In [ ]:
id(256)

10922656

In [ ]:
id(2345)

140642997747152

In [ ]:
id(21347845645)

140642997744976

In [ ]:
test = 123
id(test)

10918400

> 빈 공간 중 아무 데나 저장되기 때문에 사람에 따라 결과는 다름!

#6. 함수 지정하기(함수 만들기)

여태 우리가 수많은 예제를 하며 많이 해 온 활동임!
대신, 정의를 해 줘야 한다.

In [ ]:
def convert_to_celsius(fahrenheit):
return (fahrenheit - 32) * 5 / 9

convert_to_celsius(80)

IndentationError: ignored

> 함수를 정의했을 땐 정의 내용을 어느 정도의 간격으로 띄워줘야 됨!

In [ ]:
def convert_to_celsius(fahrenheit):
  return (fahrenheit - 32) * 5 / 9

convert_to_celsius(80)

26.666666666666668

>1. Python **executes the function definition**, which creates the function object
(but doesn’t execute it yet).
2. Next, Python **executes function** call convert_to_celsius(80). To do this, it assigns 80 to fahrenheit (which is a variable). For the duration of this function call, fahrenheit refers to 80.
3. Python now **executes the return statement**. fahrenheit refers to 80, so the
expression that appears after return is equivalent to (80 - 32) * 5 /9. WhenPython evaluates that expression, 26.666666666666668 is produced. We use the word return to tell Python what value to produce as the result of the function call, so the result of calling convert_to_celsius(80) is 26.666666666666668.
4. Once Python has finished executing the function call, it returns to the place where the function was originally called.
